src/consensus/aft/raft.h

```diff
@@ -1117,6 +1178,61 @@ namespace aft
         append_entries.push_back(std::make_tuple(std::move(ds), i));
       }
 
+      is_execution_pending = true;
+      auto msg = std::make_unique<threading::Tmsg<AsyncExecution>>(
+        execute_append_entries_cb,
+        this,
+        std::move(append_entries),
+        std::move(r),
+        confirm_evidence);
+      if (threading::ThreadMessaging::thread_count > 1)
+      {
+        threading::ThreadMessaging::thread_messaging.add_task(
+          threading::ThreadMessaging::get_execution_thread(
+            threading::MAIN_THREAD_ID),
+          std::move(msg));
+      }
+      else
+      {
+        guard.unlock();
+        msg->cb(std::move(msg));
+      }
```

Comment:
As this pattern is repeated below, could we create a new wrapper in `thread_messaging.h` that adds an async task if `thread_count > 1` and otherwise executes the callback synchronously?